### PR TITLE
Integrate node-pre-gyp

### LINKED ISSUE #4: Get node-pre-gyp working
[node-pre-gyp](https://github.com/mapbox/node-pre-gyp) is a tool for easy cross-platform binary deployment of native node.js addons. As status-nodejs requires `libstatus.a` to build a native addon, it's easier to provide binaries for various platforms hosted on S3.

### PR DESCRIPTION
**Pull request by adambabik**
This PR brings a lot of features!

1. Integration with node-pre-gyp,
2. Integration with node-pre-gyp-github,
3. Easy releases with `npm run publish` and `npm run publish:release`,
4. Updated Readme.

Closes #4.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
+    "host": "https://github.com/status-im/status-nodejs/releases/download/",
+    "remote_path": "{version}"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/status-im/status-nodejs.git"
   },
-  "author": "Status",
-  "license": "ISC",
+  "contributors": [
```
Comment (adambabik) [inline package.json:36]: @divan @flexsurfer let me know your email addresses or if you'd like to be listed here at all.

Comment (divan) [reply package.json:36]: @adambabik I don't :)

File: src/status.cpp [hunk 2]
```diff
@@ -841,35 +878,9 @@ void _NotifyUsers(const FunctionCallbackInfo<Value>& args) {
...
...ET_METHOD(exports, "CallRPC", _CallRPC);NODE_SET_METHOD(exports, "CreateAccount", _CreateAccount);NODE_SET_METHOD(exports, "CreateChildAccount", _CreateChildAccount);NODE_SET_METHOD(exports, "RecoverAccount", _RecoverAccount);NODE_SET_METHOD(exports, "VerifyAccountPassword", _VerifyAccountPassword);NODE_SET_METHOD(exports, "Login", _Login);NODE_SET_METHOD(exports, "Logout", _Logout);NODE_SET_METHOD(exports, "CompleteTransaction", _CompleteTransaction);NODE_SET_METHOD(exports, "CompleteTransactions", _CompleteTransactions);NODE_SET_METHOD(exports, "DiscardTransaction", _DiscardTransaction);NODE_SET_METHOD(exports, "DiscardTransactions", _DiscardTransactions);NODE_SET_METHOD(exports, "InitJail", _InitJail);NODE_SET_METHOD(exports, "Parse", _Parse);NODE_SET_METHOD(exports, "CreateAndInitCell", _CreateAndInitCell);NODE_SET_METHOD(exports, "ExecuteJS", _ExecuteJS);NODE_SET_METHOD(exports, "Call", _Call);NODE_SET_METHOD(exports, "StartCPUProfile", _StartCPUProfile);NODE_SET_METHOD(exports, "StopCPUProfiling", _StopCPUProfiling);NODE_SET_METHOD(exports, "WriteHeapProfile", _WriteHeapProfile);NODE_SET_METHOD(exports, "Notify", _Notify);NODE_SET_METHOD(exports, "NotifyUsers", _NotifyUsers);
```
Comment (divan) [inline src/status.cpp]: Something wrong here (diff?). All functions should be exported via NODE_SET_METHOD, no?

Comment (adambabik) [reply src/status.cpp]: Good catch! I removed a newline by accident.

File: tools/go2nodebinding/template.go [hunk 3]
```diff
@@ -61,16 +59,22 @@ void _{{.Name}}(const FunctionCallbackInfo<Value>& args) {
 
 	// Check the argument types
 	{{range $index, $arg := .Params}}
-	{{if eq .Type "*C.char"}}if (!args[{{$index}}]->IsString()) { {{end}}{{if eq .Type "C.int"}}if (!args[{{$index}}]->IsNumber()) { {{end}}
+	{{- if eq .Type "*C.char" -}}if (!args[{{$index}}]->IsString()) { {{- end}}
```
Comment (divan) [inline tools/go2nodebinding/template.go:64]: What does `-` in `{{- if` stands for?

Comment (adambabik) [reply tools/go2nodebinding/template.go:64]: Removes white spaces before expression.

File: src/status.cpp (hunk 2 above)
Comment (divan) [reply src/status.cpp]: Ah, so it's just a long line, right. Yeah, newline will do.

**Review by divan — APPROVED** [review]

**Pull request merged**